### PR TITLE
Add Proxy Support to Unix Install Script

### PR DESCRIPTION
**Pull request by schmikei**
## Description of Changes

Adds flags to the installer script

```
    -x, --proxy
        Defines the proxy server to be used for communication by the install script.
        Example: -x http(s)://server-ip:port/.

    -U, --proxy-user
        Defines the proxy user to be used for communication by the install script.

    -P, --proxy-password
        Defines the proxy password to be used for communication by the install script.
```

## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.6896461|-0.10351205|128.58633|-0.60588074|
|1|5000|5.96555|-0.22419643|137.22495|-0.3152008|
|1|10000|11.017305|-0.74163723|143.51643|-5.091461|
|1|50000|64.8459|+0.19018555|170.40355|-9.0784|
|1|100000|132.30176|+0.47203064|221.2597|-9.964844|
|10|100|2.1552083|-0.017205715|132.0198|-0.9924469|
|10|500|6.982816|-0.655272|138.3129|-0.36880493|
|10|1000|13.327756|-0.20698738|146.12352|-1.2206268|
|10|5000|68.62156|+2.0688324|173.36772|-4.232895|
|10|10000|126.41905|-6.7586517|226.74866|-1.9543304|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.6207188|-0.17243934|128.33418|-0.8580322|
|1|5000|6.310548|+0.12080145|136.86086|-0.6792908|
|1|10000|12.569058|+0.8101158|148.54324|-0.06465149|
|1|50000|63.535824|-1.1198921|176.35667|-3.1252747|
|1|100000|124.91839|-6.911339|224.70743|-6.5171204|
|10|100|2.034459|-0.13795495|132.03624|-0.9760132|
|10|500|7.1553073|-0.48278093|139.78853|+1.1068268|
|10|1000|13.241616|-0.29312706|146.12257|-1.2215729|
|10|5000|58.640026|-7.9127007|174.43777|-3.1628418|
|10|10000|130.09698|-3.080719|237.87204|+9.169052|

File: scripts/unix-install.sh [hunk 1]
```diff
@@ -535,6 +559,40 @@ pipeline:
 EOF
 }
 
+set_proxy()
+{
+      if [ -n "$proxy" ]; then
+    info "Using proxy from arguments: $proxy"
```
Comment (jsirianni) [inline scripts/unix-install.sh:565]: indent

File: scripts/unix-install.sh [hunk 2]
```diff
@@ -451,8 +463,20 @@ install_package()
...
+    fi
+  fi
+
+  if [ -n "$proxy" ]; then
+    info "Downloading package using proxy..."
+  fi 
+
   info "Downloading binary..."
-  curl -L "$agent_download_url" -o "$agent_binary" --progress-bar --fail || error_exit "$LINENO" "Failed to download package"
+  curl -L "${proxy_args}" "$agent_download_url" -o "$agent_binary" --progress-bar --fail || error_exit "$LINENO" "Failed to download package"
```
Comment (jsirianni) [inline scripts/unix-install.sh:479]: Need to use `eval` here, when inserting arguments. For example, `proxy_args` expands to `-x <proxy ip> -U username:password`. The -x and -U will not behave:

without eval
```
scripts/unix-install.sh --proxy squid:3128 --proxy-user user --proxy-password password
```
```
...
===================================================
| Configuring Installation Variables
===================================================
  Using proxy from arguments: 10.33.105.14:3128
  Configuration complete!

===================================================
| Installing Stanza
===================================================
  Creating Stanza directory...
    Succeeded!
  Checking that service is not running...
    Succeeded!
  Downloading package using proxy...
  Downloading binary...

curl: (5) Could not resolve proxy:  10.33.105.14
    ERROR (scripts/unix-install.sh:479): Failed to download package
```

With eval, I get no error and my proxy logs the connection.

```suggestion
  eval curl -L "${proxy_args}" "$agent_download_url" -o "$agent_binary" --progress-bar --fail || error_exit "$LINENO" "Failed to download package"
```

**Review by jsirianni — CHANGES_REQUESTED** [review]
You can use this docker command to run a proxy locally
```
docker run \
  -p 3128:3128 \
  bmedora/squid:0.1.2 \
  /apps/squid/sbin/squid -NsY -f /apps/squid.conf.basicauth
```
```
scripts/unix-install.sh --proxy localhost:3128 --proxy-user user1 --proxy-password user1
```

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.655216|-0.1379422|131.1584|+1.9661865|
|1|5000|5.637965|-0.5517812|135.01495|-2.5251923|
|1|10000|11.775937|+0.016994476|148.0058|-0.60209656|
|1|50000|60.19091|-4.4648056|172.77762|-6.7043304|
|1|100000|137.15738|+5.327652|235.99596|+4.771408|
|10|100|2.017221|-0.15519309|132.83647|-0.17578125|
|10|500|7.207092|-0.43099642|139.78004|+1.0983429|
|10|1000|13.017385|-0.5173588|146.62096|-0.7231903|
|10|5000|68.25973|+1.7070007|180.55226|+2.951645|
|10|10000|144.79688|+11.619171|234.81493|+6.1119385|

**Review by jsirianni — APPROVED** [review]

**Pull request merged**